### ISSUE TITLE
`append()` with `lockmarks` causes invalid buffer updates

### ISSUE THREAD

**Issue by vigoux** [neovim/neovim#14772]
While debugging [this issue](https://github.com/neovim/neovim/issues/12861#issuecomment-857624004) I ended bisecting the issue to [this line](https://github.com/prettier/vim-prettier/blob/aa0607ca7a0f61e91365ecf25947312ff4796302/autoload/prettier/utils/buffer.vim#L23) in `vim-prettier`.

I am not sure that [this other line](https://github.com/prettier/vim-prettier/blob/aa0607ca7a0f61e91365ecf25947312ff4796302/autoload/prettier/utils/buffer.vim#L18) causes issues too.

The issue looks a bit worrying.

- `nvim --version`: master
- Operating system/version: Void Linux
- Terminal name/version: st
- `$TERM`: not relevant

[ ] `vim -u DEFAULTS` (version: ) behaves differently

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
# Set tree-sitter up
:lockmarks call append(1, "abc")
```

### Actual behaviour

Errors appear using tree-sitter (symptom for invalid byte-level updates).

### Expected behaviour

No errors

### Additional informations

I am sure this is reproducible with the `append` function.
My fear is that this issue is generic, as long as `lockmarks` is on.

**Comment by vigoux** [comment]
I added a test to the buffer updates specs, and the bytecount is indeed wrong.

**Comment by vigoux** [comment]
The true question is then `should extmarks move when :lockmarks ?`.

I'd say yes, but I think the true expert here is @bfredl.

**Comment by bfredl** [comment]
Yes, they should. extmarks move with the text, not with arbitrary design desicions of vim marks.
